odoo/addons/base/models/ir_model.py

```diff
@@ -470,10 +470,12 @@ def _related_field(self):
                 raise UserError(_("Unknown field name '%s' in related field '%s'") % (name, self.related))
             if index < last and not field.relational:
                 raise UserError(_("Non-relational field name '%s' in related field '%s'") % (name, self.related))
+            if self.store and not field.store and not field.compute:
+                raise UserError(_("Field %s (%s) cannot be used in stored related field.") % (field.string, field.name))
             model = model[name]
         return field
 
-    @api.constrains('related')
+    @api.constrains('related', 'store')
```

Comment:
I don't get that part.

Reply:
_check_related calls _related_field which now checks `self.store`